pinot-segment-local/src/test/java/org/apache/pinot/segment/local/recordtransformer/ComplexTypeTransformerTest.java

```diff
@@ -237,4 +238,76 @@ public void testUnnestCollection() {
     next = itr.next();
     Assert.assertEquals("v2", next.getValue("array.a"));
   }
+
+  @Test
+  public void testConvertCollectionToString() {
+    // json convert inner collections
+    // {
+    //   "array":[
+    //      {
+    //         "array1":[
+    //            {
+    //               "b":"v1"
+    //            }
+    //         ]
+    //      }
+    //   ]
+    // }
+    // is converted to
+    // [{
+    //   "array.array1":"[
+    //            {
+    //               "b":"v1"
+    //            }
+    //         ]"
+    // }]
+    ComplexTypeTransformer transformer = new ComplexTypeTransformer(Arrays.asList("array"), ".");
+    GenericRow genericRow = new GenericRow();
+    Map<String, Object> map = new HashMap<>();
+    Object[] array1 = new Object[1];
+    array1[0] = ImmutableMap.of("b", "v1");
+    map.put("array1", array1);
+    Object[] array = new Object[1];
+    array[0] = map;
+    genericRow.putValue("array", array);
+    transformer.transform(genericRow);
+    Assert.assertNotNull(genericRow.getValue(GenericRow.MULTIPLE_RECORDS_KEY));
+    Collection<GenericRow> collection = (Collection<GenericRow>) genericRow.getValue(GenericRow.MULTIPLE_RECORDS_KEY);
+    GenericRow row = collection.iterator().next();
+    Assert.assertTrue(row.getValue("array.array1") instanceof String);
+
+    // primitive array not converted
```

Comment:
The rules aren't clear. As per the design doc (in the Collection handling section), if collection unnesting rule is not applied on a collection, then it will use JSON serialization rule right?
So why are we differentiating now on primitive collection v/s collection of non scalars ?

Reply:
coz Pinot natively support multi-value column. Also, per @Jackie-Jiang's suggestion, I'll add a mode so user can choose JSON serialize all collections or collections of non-primitive values.

Reply:
Yes but native multi-value column support for primitives should not be confused with array/collection inside a JSON object right even though the array/collection inside a JSON object could be of primitives. These two are completely different

Reply:
then how is an array of integers in Avro ingested into Pinot, today?

Reply:
As a MV column. But here we are talking about array inside a nested object which doesn't get the same storage format and treatment (at least in the current design) as an MV column in the schema. 

If I have something like 

```
{
   name
   age
   salary
   array of organizationIds (primitive integers)
   array of my addresses (another nested struct)
}
```

Even though array of organizationIds is an array of primitive INTs, we currently don't store it in the same way as we would have done if it were an independent MV column in the schema. So the collection unnesting rule we have in design should be applicable to both collections in the same way. Similarly, I think if the user chooses to not unnest them then the behavior should be same for both collections regardless of whether one is for primitives and other is not.

Reply:
I see. The disconnect here is that there is no nested collection. All nested collection will be flattened to become top-level, and we have two choices on them: collections of pritimive values stay the same while collections of non-primitives convert to String